src/corvus.c

```diff
@@ -18,14 +18,33 @@
 #include "timer.h"
 #include "alloc.h"
 
+static pthread_spinlock_t signal_lock;
 static struct context *contexts;
 
-void sig_handler(int sig)
+void sigsegv_handler(int sig)
 {
-    int i;
+    if (pthread_spin_trylock(&signal_lock) == EBUSY) {
+        sigset_t mask;
+        sigemptyset(&mask);
+        pselect(0, NULL, NULL, NULL, NULL, &mask);
```

Comment:
@maralla If what we intended is to ignore the following `SIGSEGV`, we could just `return` when the lock is busy.